src/js/main/index.js

```diff
@@ -0,0 +1,42 @@
+import docsearch from "docsearch.js";
+import AnchorJS from "anchor-js";
+import ImageObserver from "./image-lazy-loader";
+
+const htmlClassList = document.documentElement.classList;
+
+htmlClassList.remove("no-js");
+htmlClassList.add("js");
+
+// Initialize DocSearch
+docsearch({
+    apiKey: "891b0e977d96c762a3821e0c00172ac9",
+    indexName: "eslint",
+    inputSelector: "#eslint-search-input",
+    algoliaOptions: { facetFilters: [["tags:docs", "tags:blog"]] },
+    debug: false // Set debug to true if you want to inspect the dropdown
+});
+
+document.addEventListener("DOMContentLoaded", function() {
+
+    // Initialize deep anchor links for headings
+    const anchors = new AnchorJS();
+
+    anchors.add(".doc h2, .doc h3, .doc h4, .blog h2, .blog h3, .blog h4");
+
+    // Initialize lazy loading images
+    ImageObserver.createObserver();
```

Comment:
Load this on every page since we also attempt to lazy load images in the blog as well.